bin/kano-sync

```diff
@@ -183,49 +204,64 @@ def do_restore():
     _, e, _ = run_cmd(tar_cmd)
     if e:
         msg = 'Error with uncompressing restore data: {}'.format(e)
-        display_msg(msg, 'error')
+        display_msg(msg, 'error', dialog=dialog)
 
-    refresh_kdesk()
+    if not skip_kdesk_refresh:
+        refresh_kdesk()
 
     delete_file(tmp_file)
     delete_dir(tmp_dir)
 
-    display_msg('Restore OK', 'info')
+    display_msg('Restore OK', 'info', dialog=dialog)
 
 
-def do_upload_tracking_data():
+def do_upload_tracking_data(dialog=True):
     logger.debug('do_upload_tracking_data')
 
     success, value = upload_tracking_data()
     if not success:
-        display_msg('Upload not possible, error: {}'.format(value), 'error')
+        display_msg(
+            'Upload not possible, error: {}'.format(value),
+            'error',
+            dialog=dialog
+        )
         return
 
 
 def refresh_kdesk():
-    if args.skip_kdesk:
-        return
-
     logger.info('refreshing kdesk from kano-sync')
     if os.path.exists('/usr/bin/kdesk'):
         run_bg('kdesk -a profile ; kdesk -a world')
 
 
-if __name__ == '__main__':
-    if not (args.sync or args.backup or args.restore or args.upload_tracking_data):
-        parser.print_help()
-        sys.exit()
+def main(args):
```

Comment:
It would be nice to see everything else migrated out of here into a different module so this just does the parsing but it is just a nice to have.